rust/kernel/c_types.rs

```diff
@@ -51,3 +51,14 @@ mod c {
 }
 
 pub use c::*;
+
+/// Reads string until null byte is reached and returns slice excluding the terminating null.
+///
+/// SAFETY: The data from the pointer until the null terminator must be valid for reads and
+/// not mutated for all of `'a`. The length of the string must also be less than `isize::MAX`.
+/// See the documentation on [`from_raw_parts`](https://doc.rust-lang.org/core/slice/fn.from_raw_parts.html)
+/// for further details on safety of converting a pointer to a slice.
```

Comment:
`SAFETY: ...` comments are used for explaining why some unsafe code is actually safe; however, to explain what are the additional constraints on callers of an `unsafe` function, a `# Safety` section is used in the function docs. Also, it seems the standard library uses out-of-line links (I also reduced the length of the lines a bit, otherwise it seem it overflow GitHub too and the standard library also seems to use 80 as the limit for comments).

```suggestion
/// # Safety
///
/// The data from the pointer until the null terminator must be valid for reads
/// and not mutated for all of `'a`. The length of the string must also be less
/// than `isize::MAX`. See the documentation on [`from_raw_parts`] for further
/// details on safety of converting a pointer to a slice.
///
/// [`from_raw_parts`]: https://doc.rust-lang.org/core/slice/fn.from_raw_parts.html
```